spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/FluxMessageChannelBindingTargetFactory.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binding;
+
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.integration.channel.FluxMessageChannel;
+
+/**
+ * @author Soby Chacko
+ * @since 4.0.0
+ */
+public class FluxMessageChannelBindingTargetFactory extends AbstractBindingTargetFactory<FluxMessageChannel> {
+
+	private final MessageChannelConfigurer messageChannelConfigurer;
+
+	@Autowired
+	private GenericApplicationContext context;
```

Comment:
IMHO it is always bad to use `@Autowired`: you are loosing context of dependency injection.
Plus `@Autowired` is based on reflection, so better to rely on respective Spring callbacks.
It is probably OK for demos and target projects, but I'd prefer to stay away from magical injections in the framework.

Reply:
Ok, will change to constructor injection.